lightning/src/routing/router.rs

```diff
@@ -876,39 +877,89 @@ pub fn get_route<L: Deref>(our_node_id: &PublicKey, network: &NetworkGraph, paye
 		// If a caller provided us with last hops, add them to routing targets. Since this happens
 		// earlier than general path finding, they will be somewhat prioritized, although currently
 		// it matters only if the fees are exactly the same.
-		for hop in last_hops.iter() {
+		for route in last_hops.iter() {
+			let first_hop_in_route = &(route.0)[0];
```

Comment:
Including a simple `if` check to make sure no empty `RouteHint`s are considered.

Reply:
Commit [9d0b1](https://github.com/rust-bitcoin/rust-lightning/pull/1040/commits/9d0b17971bae409429f9750ba479f5c5499de414) does this.

Reply:
Could you use `filter` instead to avoid adding another level of indenting?

```rust
for route in last_hops.iter().filter(|route| !route.0.is_empty()) {
```